Bottom: bdbfa12f9ea1ca21250c55fdfcf447d3a2c80cd0
Top:    bceba743d1c1de7d9da4c6edadee0839e2455f2e
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:57 -0500

Move git checkouts to respective roles


---
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index 85be0141..da75e132 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -34,24 +34,6 @@
     group: root
     mode: 'u=rwx,g=rx,o=rx'
 
-- name: docker projects base directory
-  file:
-    state: directory
-    path: '{{ docker_projects_dir }}'
-    owner: root
-    group: root
-    mode: 'u=rwx,g=rx,o=rx'
-
-- name: checkout git repositories
-  git:
-    repo: '{{ item.repo }}'
-    dest: '{{ docker_projects_dir }}/{{ item.name }}'
-    clone: yes
-    recursive: yes
-    update: yes
-  with_items: "{{ docker_projects }}"
-  tags:
-    - docker_repositories
 
 - name: Check if baculavirt selinux module exists
   stat:
diff --git a/roles/docker-common/vars/main.yml b/roles/docker-common/vars/main.yml
index eab14eae..b4b1526c 100644
--- a/roles/docker-common/vars/main.yml
+++ b/roles/docker-common/vars/main.yml
@@ -2,25 +2,6 @@
 
 #docker_project_repository: https://github.com/JosephLutz/docker-project.git
 
-# the git repositoryies to check out for building the docker containers
-docker_projects:
-  - name: docker-openldap
-    repo: https://github.com/novatechweb/docker-openldap.git
-  - name: docker-mediawiki
-    repo: https://github.com/novatechweb/docker-mediawiki.git
-  - name: docker-gitlab
-    repo: https://github.com/novatechweb/docker-gitlab.git
-  - name: docker-openssl
-    repo: https://github.com/novatechweb/docker-openssl.git
-  - name: docker-svn
-    repo: https://github.com/novatechweb/docker-svn.git
-  - name: docker-mantisbt
-    repo: https://github.com/novatechweb/docker-mantisbt.git
-  - name: docker-mysql-data
-    repo: https://github.com/novatechweb/docker-mysql-data.git
-  - name: docker-exim4
-    repo: https://github.com/novatechweb/docker-exim4.git
-
 # the base directory where the docker projects will be checked out
 docker_projects_dir: '/opt/docker/{{ docker_image_tag }}'
 docker_restore_config_base_dir: '/etc/docker_container.conf/{{ docker_image_tag }}'
diff --git a/roles/docker-exim4/tasks/main.yml b/roles/docker-exim4/tasks/main.yml
index 0c155b43..49eccc30 100644
--- a/roles/docker-exim4/tasks/main.yml
+++ b/roles/docker-exim4/tasks/main.yml
@@ -76,6 +76,12 @@
 # *****************************************************************************
 # Update or make the image.
 
+- name: Checkout image repo
+  git:
+    repo: '{{ exim4_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-exim4'
+
 - name: build image
   docker_build:
     image_name: '{{ exim4_image_name }}'
diff --git a/roles/docker-exim4/vars/main.yml b/roles/docker-exim4/vars/main.yml
index ef878299..20d48fc4 100644
--- a/roles/docker-exim4/vars/main.yml
+++ b/roles/docker-exim4/vars/main.yml
@@ -1,5 +1,8 @@
 ---
 
+# the repository containing Dockerfile to build the image
+exim4_image_repo: https://github.com/novatechweb/docker-exim4.git
+
 # the name of the image being duilt and used for the container
 exim4_image_name: '{{ docker_registry_username }}/exim4'
 
diff --git a/roles/docker-gitlab/tasks/main.yml b/roles/docker-gitlab/tasks/main.yml
index 8404097a..aeab7bf9 100644
--- a/roles/docker-gitlab/tasks/main.yml
+++ b/roles/docker-gitlab/tasks/main.yml
@@ -74,6 +74,12 @@
 # *****************************************************************************
 # Update or make the image.
 
+- name: Checkout image repo
+  git:
+    repo: '{{ gitlab_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-gitlab'
+
 - name: build image
   docker_build:
     image_name: '{{ gitlab_image_name }}'
diff --git a/roles/docker-gitlab/vars/main.yml b/roles/docker-gitlab/vars/main.yml
index b246e5fd..91620987 100644
--- a/roles/docker-gitlab/vars/main.yml
+++ b/roles/docker-gitlab/vars/main.yml
@@ -1,6 +1,7 @@
 ---
 
 # the name of the image being duilt and used for the container
+gitlab_image_repo: 'https://github.com/novatechweb/docker-gitlab.git'
 gitlab_image_name: '{{ docker_registry_username }}/gitlab'
 gitlab_dv_image_name: '{{ docker_registry_username }}/gitlab'
 gitlab_db_image_name: '{{ repository_images.postgres.image }}'
diff --git a/roles/docker-mantisbt/tasks/main.yml b/roles/docker-mantisbt/tasks/main.yml
index 1f1e817c..7d2c6898 100644
--- a/roles/docker-mantisbt/tasks/main.yml
+++ b/roles/docker-mantisbt/tasks/main.yml
@@ -70,6 +70,11 @@
 
 # *****************************************************************************
 # Update or make the image.
+- name: Checkout image repo
+  git:
+    repo: '{{ mantisbt_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-mantisbt'
 
 - name: build image
   docker_build:
@@ -77,6 +82,12 @@
     image_tag: '{{ docker_image_tag }}'
     dockerfile_dir: '{{ docker_projects_dir }}/docker-mantisbt'
 
+- name: Checkout db repo
+  git:
+    repo: '{{ mantisbt_db_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-mysql-data'
+
 - name: build data-container image
   docker_build:
     image_name: '{{ mantisbt_db_image_name }}'
diff --git a/roles/docker-mantisbt/vars/main.yml b/roles/docker-mantisbt/vars/main.yml
index 7dbe8249..fac5a9c8 100644
--- a/roles/docker-mantisbt/vars/main.yml
+++ b/roles/docker-mantisbt/vars/main.yml
@@ -1,7 +1,10 @@
 ---
 
 # the name of the image being duilt and used for the container
+mantisbt_image_repo: https://github.com/novatechweb/docker-mantisbt.git
 mantisbt_image_name: '{{ docker_registry_username }}/mantisbt'
+
+mantisbt_db_image_repo: https://github.com/novatechweb/docker-mysql-data.git
 mantisbt_db_image_name: '{{ docker_registry_username }}/mysql.dv'
 
 # the name of the container being started
diff --git a/roles/docker-mediawiki/tasks/main.yml b/roles/docker-mediawiki/tasks/main.yml
index dabe61e1..3dda287b 100644
--- a/roles/docker-mediawiki/tasks/main.yml
+++ b/roles/docker-mediawiki/tasks/main.yml
@@ -71,6 +71,12 @@
 # *****************************************************************************
 # Update or make the image.
 
+- name: Checkout image repo
+  git:
+    repo: '{{ wiki_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-mediawiki'
+
 - name: build image
   docker_build:
     image_name: '{{ wiki_image_name }}'
diff --git a/roles/docker-mediawiki/vars/main.yml b/roles/docker-mediawiki/vars/main.yml
index c5844050..c4a8db82 100644
--- a/roles/docker-mediawiki/vars/main.yml
+++ b/roles/docker-mediawiki/vars/main.yml
@@ -1,5 +1,7 @@
 ---
 
+wiki_image_repo: https://github.com/novatechweb/docker-mediawiki.git
+
 # the name of the image being duilt and used for the container
 wiki_image_name: '{{ docker_registry_username }}/wiki'
 wiki_db_image_name: '{{ repository_images.mysql.image }}'
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 51ceb46b..332dba7b 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -60,6 +60,12 @@
 
 # *****************************************************************************
 # Update or make the image.
+- name: Checkout image repo
+  git:
+    repo: '{{ openldap_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-openldap'
+
 
 - name: build image
   docker_build:
diff --git a/roles/docker-openldap/vars/main.yml b/roles/docker-openldap/vars/main.yml
index e4317955..516cb9a8 100644
--- a/roles/docker-openldap/vars/main.yml
+++ b/roles/docker-openldap/vars/main.yml
@@ -1,6 +1,7 @@
 ---
 
 # the name of the image being duilt and used for the container
+openldap_image_repo: https://github.com/novatechweb/docker-openldap.git
 openldap_image_name: '{{ docker_registry_username }}/ldap'
 
 # the name of the container being started
diff --git a/roles/docker-openssl/tasks/main.yml b/roles/docker-openssl/tasks/main.yml
index 2c0f093b..05b9a20d 100644
--- a/roles/docker-openssl/tasks/main.yml
+++ b/roles/docker-openssl/tasks/main.yml
@@ -54,6 +54,12 @@
 # *****************************************************************************
 # Update or make the image.
 
+- name: Checkout image repo
+  git:
+    repo: '{{ openssl_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-openssl'
+
 - name: build image
   docker_build:
     image_name: '{{ openssl_image_name }}'
diff --git a/roles/docker-openssl/vars/main.yml b/roles/docker-openssl/vars/main.yml
index 31679cda..7e661a4b 100644
--- a/roles/docker-openssl/vars/main.yml
+++ b/roles/docker-openssl/vars/main.yml
@@ -1,6 +1,7 @@
 ---
 
 # the name of the image being duilt and used for the container
+openssl_image_repo: https://github.com/novatechweb/docker-openssl.git
 openssl_image_name: '{{ docker_registry_username }}/openssl'
 
 # the name of the data-volume used by the container
diff --git a/roles/docker-svn/tasks/main.yml b/roles/docker-svn/tasks/main.yml
index fffe49a5..a34e61d0 100644
--- a/roles/docker-svn/tasks/main.yml
+++ b/roles/docker-svn/tasks/main.yml
@@ -53,6 +53,11 @@
 
 # *****************************************************************************
 # Update or make the image.
+- name: Checkout image repo
+  git:
+    repo: '{{ svn_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/docker-svn'
 
 - name: build image
   docker_build:
diff --git a/roles/docker-svn/vars/main.yml b/roles/docker-svn/vars/main.yml
index 7d02a48c..f8e412d3 100644
--- a/roles/docker-svn/vars/main.yml
+++ b/roles/docker-svn/vars/main.yml
@@ -1,6 +1,7 @@
 ---
 
 # the name of the image being duilt and used for the container
+svn_image_repo: https://github.com/novatechweb/docker-svn.git
 svn_image_name: '{{ docker_registry_username }}/svn'
 
 # the name of the container being started
@@ -12,4 +13,3 @@ svn_dv_name: '{{ docker_name_prefix }}svn_DV'
 # restore directories to temporarly store data being restored into docker containers
 svn_docker_backup_dir: '{{ docker_backup_dir }}/SVN'
 host_svn_docker_restore_dir: '{{ bacula_dest }}{{ svn_docker_backup_dir }}'
-
